### PR TITLE
General cleanup and prep for TailwindCSS

### PR DESCRIPTION
**Pull request by tony-sull**
⚠️ This ended up being a bit of a large PR based on files touched, sorry!

- moved component-specific styles out of `main.css`
- removed Prism styles now that we're using Shiki
- refactored page layouts (a bit of duplicate code cleanup, should help when adding Tailwind)
- moved all scripts into `src`, refactored them a bit, and moved component-specific scripts out of the base layout

### CODE REVIEW

File: src/components/PixelLink.astro [hunk 1]
```diff
@@ -6,6 +6,13 @@ const { variant = 'primary' } = Astro.props;
 const { before, after } = Astro.slots;
 ---
 
+<!-- Blocking script -->
+<script global hoist>
```
Comment (natemoo-re) [inline src/components/PixelLink.astro:10]: Does `global` do anything?

**Review by natemoo-re — APPROVED** [review]
This looks amazing, great work!

File: src/components/PixelLink.astro (hunk 1 above)
Comment (tony-sull) [reply src/components/PixelLink.astro]: Great catch, it doesn't!  I was trying out a bunch of different script setups and didn't notice this had both flags :+1:

**Comment by FredKSchott** [comment]
Hate to do this but there were some bugs in the blog that broke some posts, including the v0.23 release post. I just reverted the commit, if you can resolve the issues in #35 then I think this PR can be re-merged!

**Pull request merged**